[docs] Update README.md to match Introduction section of the docs (#5754)
mui/mui-x: README.md

<h1 align="center">MUI X</h1>

MUI X is a collection of advanced UI components, including a [data grid](https://mui.com/x/react-data-grid/) and [date and time pickers](https://mui.com/x/react-date-pickers/getting-started/).

These components are significantly more complex than the ones found in the [MUI Core](https://github.com/mui/material-ui/) libraries.
They feature advanced functionality for data-rich applications and a wide range of other use cases.

MUI X is **open core**—base components are MIT-licensed, while more advanced features require a Pro or Premium commercial license.
See the [Licensing page](https://mui.com/x/introduction/licensing/) for details.

## Components

> - [Data Grid](https://mui.com/x/react-data-grid/)
> - [Date and Time Pickers](https://mui.com/x/react-date-pickers/getting-started/)

Reviewer: In the context of our docs?

```suggestion
- [Data grid](https://mui.com/x/react-data-grid/)
- [Date and Time pickers](https://mui.com/x/react-date-pickers/getting-started/)
```

## Installation

> ### Data Grid

Reviewer: ```suggestion
### Data grid
```

Read the Data Grid [Installation instructions](https://mui.com/x/react-data-grid/getting-started/#installation) in the MUI X docs.

> ### Date and Time Pickers

Reviewer: ```suggestion
### Date and Time pickers
```

Author: This kind of capitalization looks very awkward to me. Is it title case? Is it sentence case? It seems to be somewhere in between. I have proposed here https://github.com/mui/mui-x/issues/5764#issuecomment-1213536072 that we refer to Date and Time Pickers as a proper noun, always capitalized, unless referring to the general concept of these components. Same for Data Grid—it seems weird to me that we wouldn't treat our product names as Proper Nouns in title case. It would be like writing "MUI base."

Reviewer 2: I'd also vote for going with `title case` for component names, but that would mean replacing all of them in both `mui` and `mui-x` repositories and in such case we need to decide if it's even worth it..? 🤔

Author: I think it's worth it for the sake of consistency and professionalism. It's something I've begun to address as I revise individual pages in #5764—see #5780 and #5781—but I haven't moved forward with it since we still don't have a consensus on this point.

Reviewer: @samuelsycamore I didn't know we started a discussion on changing this convention. I have been making changes for a couple of months in the opposite direction, e.g. in Material UI https://github.com/mui/material-ui/pull/32238 😁. I think that my rationale was:

1. sentence case is simpler
2. used by MD https://material.io/components/data-tables
3. It makes it easier to understand if you mean the component data grid or the React component DataGrid (Data Grid being visually closer, hence more confusing)

Now, I do agree, when you look at the Google search results, you see the Title case. Google is changing the capitalization (caught by ahrefs) of the pages listed.

No objections from my end, as long as we update everything. cc @mbrookes maybe for thoughts as well.

Author: Let's do it!

Reviewer: OK, so we make a new exception to our rule. We always use sentence case as the default, but we use Title Case for roles, e.g. Developer Advocate, and for component names, e.g. Text Field.

> Read the Date and Time Pickers [Installation instructions](https://mui.com/x/react-date-pickers/getting-started/#setup) in the MUI X docs.

Reviewer: ```suggestion
Read the Date and Time pickers [Installation instructions](https://mui.com/x/react-date-pickers/getting-started/#setup) in the MUI X docs.
```

The Premium version is available under a commercial license—visit [the Pricing page](https://mui.com/pricing/) for details.

Premium package:

- [`@mui/x-data-grid-premium`](https://www.npmjs.com/package/@mui/x-data-grid-premium)

## Questions and feedback

### GitHub

We use GitHub issues as a bug and feature request tracker.
If you think you've found a bug, or you have a new feature idea, please start by [making sure it hasn't already been reported or fixed](https://github.com/mui/mui-x/issues?utf8=%E2%9C%93&q=is%3Aopen+is%3Aclosed).
You can search through existing issues and pull requests to see if someone has reported one similar to yours.

[Open an issue](https://github.com/mui/mui-x/issues/new/choose) in the MUI X repo.

### Stack Overflow

Visit Stack Overflow to ask questions and read crowdsourced answers from expert developers in the MUI community, as well as MUI maintainers.

[Post a question about MUI X](https://stackoverflow.com/questions/tagged/mui) on Stack Overflow using the "mui" tag.

### Bugs or feature request

You can create an [issue](https://github.com/mui/mui-x/issues) on this repository.

> Please always provide a reproduction case. The following Data Grid templates can be a good start:
> 
> - [DataGrid community version](https://codesandbox.io/s/datagrid-v5-quick-start-54iz1)
> - [DataGridPro commercial version](https://codesandbox.io/s/datagridpro-v5-quick-start-5pjhc)
> - [DataGridPremium commercial version](https://codesandbox.io/s/datagridpremium-v5-quick-start-h2ztp2)

Reviewer: Would it make sense to remove the duplication? It's covered in https://github.com/mui/mui-x/blob/master/.github/ISSUE_TEMPLATE/1.bug.yml, avoiding out-of-date links. It would also keep the message shorter, hence easier to impress the message.

```suggestion
Please always provide a reproduction case.
```

## Contributing

Read the [contributing guide](/CONTRIBUTING.md) to learn about our development process, how to propose bug fixes and improvements, and how to build and test your changes.

Contributing to MUI X is about more than just issues and pull requests!
There are many other ways to [support MUI](https://mui.com/material-ui/getting-started/faq/#mui-is-awesome-how-can-i-support-the-project) beyond contributing to the code base.

## Changelog

The [changelog](https://github.com/mui/mui-x/releases) is regularly updated to reflect what's changed in each new release.

## Roadmap

> Future plans and high-priority features and enhancements can be found in our [roadmap](https://mui.com/material-ui/discover-more/roadmap/).

Reviewer 3: ```suggestion
Future plans and high-priority features and enhancements can be found in our [roadmap](https://mui.com/x/introduction/roadmap/).
```

Link to MUI X roadmap instead?

Author: Whoops! 😅

Reviewer: I didn't know we had these two different roadmap pages. https://mui.com/material-ui/discover-more/roadmap/#new-components talks about what the next components planned for MUI X. 

I guess we need a central place to talk about the next component (not necessarily on the docs, it could be private in notion) so that Core and X don't say they will both build it.

How should we resolve this?

Author: @oliviertassinari this sounds like another page that would be best suited for a companywide document—https://github.com/mui/material-ui/issues/33514.
